### PR TITLE
New laser renderer based on signed distance fields

### PR DESCRIPTION
**Pull request by Akaricchi**
Work in progress. It is not as fast as the old renderer, but it looks much better and more consistent, doesn't suffer from aliasing artifacts, doesn't require blurring, and the appearance is a lot more customizable. Specialized shape shaders have been removed, as they don't make a great performance difference and they aren't compatible with the new curve simplification and culling. Ability to customize the coloring shader per-laser may be added easily as needed.

### CODE REVIEW

File: src/laserdraw.c [hunk 1]
```diff
@@ -0,0 +1,670 @@
...
+
+	// Vector from A to B of the last included segment, and its squared length.
+	cmplxf v0;
+	float v0_abs2;
+
+	// FIXME: we have to initialize these with something sane…
+	// but this method always causes the second sample to be skipped.
+	v0 = l->prule(l, t) - a;
+	v0_abs2 = cabs2f(v0);
+
```
[the remark this answers is not in the document]
Comment (Akaricchi) [reply src/laserdraw.c:433]: Not sure what you meant with the first proposal. `a` is already initialized with `l->rule(l, t0)` a few lines above, and then `t` is shifted, basically just like in your snippet. And we really don't want the starting point to be sampled from anything other than `t0`…

> set `v0 = a- l->prule(l, t0 - sp.time_step)`

That might work I guess? I think I'll need to test some more edge cases though, e.g. degenerate segments and extremely short straight line lasers… Even without changing anything I suspect the current code might not handle them well.
